workflow-ui/compose/src/main/java/com/squareup/workflow1/ui/compose/WorkflowRendering.kt

```diff
@@ -0,0 +1,201 @@
+        // "Fast" path: If the child factory is also a composable, we don't need to go through the
+        // legacy view system and can just compose the factory's content directly.
+        Box(modifier, propagateMinConstraints = true) {
+          viewFactory.Content(rendering, viewEnvironment)
+        }
+      } else {
+        // "Slow" path: Create a legacy Android View to show the rendering, like WorkflowViewStub.
+        ViewFactoryAndroidView(viewFactory, rendering, viewEnvironment, modifier)
+      }
+    }
+  }
+}
+
+/**
+ * Returns a [LifecycleOwner] that is a mirror of the current [LocalLifecycleOwner] until this
+ * function is skipped (no longer composed). Similar to [WorkflowLifecycleOwner] for views, but a
+ * bit simpler since we don't need to worry about attachment state.
+ */
+@Composable private fun rememberChildLifecycleOwner(): LifecycleOwner {
+  val lifecycleOwner = remember {
+    object : LifecycleOwner {
+      val registry = LifecycleRegistry(this)
+      override fun getLifecycle(): Lifecycle = registry
+    }
+  }
+  val parentLifecycle = LocalLifecycleOwner.current.lifecycle
+
+  DisposableEffect(parentLifecycle) {
+    val observer = LifecycleEventObserver { _, event ->
+      // Any time the parent lifecycle changes state, perform the same change on our lifecycle.
+      lifecycleOwner.registry.handleLifecycleEvent(event)
+    }
+
+    parentLifecycle.addObserver(observer)
+    onDispose {
+      parentLifecycle.removeObserver(observer)
+
+      // If we're being skipped it means the WorkflowRendering is either going away itself or about
+      // to switch to an incompatible rendering – either way, this lifecycle is dead.
+      // Note that we can't transition from INITIALIZED to DESTROYED – the LifecycelRegistry will
+      // throw. WorkflowLifecycleOwner has this same check.
+      if (lifecycleOwner.registry.currentState != INITIALIZED) {
+        lifecycleOwner.registry.currentState = DESTROYED
+      }
+    }
+  }
+
+  return lifecycleOwner
+}
+
+/**
+ * This is effectively the logic of [WorkflowViewStub], but translated into Compose idioms.
+ * This approach has a few advantages:
+ *
+ *  - Avoids extra custom views required to host `WorkflowViewStub` inside a Composition. Its trick
+ *    of replacing itself in its parent doesn't play nicely with Compose.
+ *  - Allows us to pass the correct parent view for inflation (the root of the composition).
+ *  - Avoids `WorkflowViewStub` having to do its own lookup to find the correct [ViewFactory], since
+ *    we already have the correct one.
+ *  - Propagate the current [LifecycleOwner] from [LocalLifecycleOwner] by setting it as the
+ *    [ViewTreeLifecycleOwner] on the view.
+ *
+ * Like `WorkflowViewStub`, this function uses the [viewFactory] to create and memoize a [View] to
+ * display the [rendering], keeps it updated with the latest [rendering] and [viewEnvironment], and
+ * adds it to the composition.
```

Comment:
I've been reflecting about whether or not we don't want to start all this kdoc with "like `WorkflowViewStub`", "like `AndroidViewRendering`" etc. as we may want to include that as an addendum after describing this on its own terms. This may be the new normal after all. Not something needed for now, but we need to consider it.

Reply:
Yea, that thought definitely crossed my mind too. I'd cross-reference back as well, but the core-android module doesn't know about this one.